src/Layer/GeoidLayer.js

```diff
@@ -0,0 +1,66 @@
+import Layer from 'Layer/Layer';
+import LayerUpdateState from 'Layer/LayerUpdateState';
+
+
+class GeoidLayer extends Layer {
+    constructor(id, config = {}) {
+        super(id, config);
+        this.isGeoidLayer = true;
+    }
+
+    update(context, layer, node, parent) {
+        if (!parent || !node.material) {
+            return;
+        }
+
+        // Don't update tile if its zoom is not within the layer's zoom limits
+        const extentsDestination = node.getExtentsByProjection(layer.crs);
+        const zoom = extentsDestination[0].zoom;
+        if (zoom > layer.zoom.max || zoom < layer.zoom.min) {
+            return;
+        }
+
+        if (node.layerUpdateState[layer.id] === undefined) {
+            node.layerUpdateState[layer.id] = new LayerUpdateState();
+        }
+
+        if (
+            layer.frozen
+            || !node.material.visible
+            || !node.layerUpdateState[layer.id].canTryUpdate()
+        ) {
+            return;
+        }
+
+        node.layerUpdateState[layer.id].newTry();
+        const command = {
+            view: context.view,
+            layer,
+            extentsSource: extentsDestination,
+            requester: node,
+        };
+
+        return context.scheduler.execute(command).then(
+            (result) => {
+                result = result[0];
+
+                if (result.data) {
+                    let geoidHeight = 0;
+                    for (const NSCardinal of ['north', 'south']) {
+                        for (const WECardinal of ['west', 'east']) {
+                            geoidHeight += result.getGeoidAtLatLon(node.extent[NSCardinal], node.extent[WECardinal]);
+                        }
+                    }
+                    geoidHeight /= 4;
+
+                    node.setBBoxGeoid(geoidHeight);
+                    node.material.geoid_height = geoidHeight;
+                    node.layerUpdateState[layer.id].noMoreUpdatePossible();
+                }
+            },
+        );
+    }
```

Comment:
could you remove command and get data directly?

Reply:
I changed this part. I now also compute the geoid height at the center of the tile extent, instead of computing the mean value of its four corners.